src/script/interpreter.cpp

```diff
@@ -2061,6 +2062,70 @@ uint256 ComputeTaprootMerkleRoot(Span<const unsigned char> control, const uint25
     return k;
 }
 
+bool VerifyAnnex(const std::vector<unsigned char>& annex_stack, ScriptExecutionData& execdata)
+{
+    CDataStream annex(MakeByteSpan(annex_stack), ANNEX_SER_TYPE, ANNEX_SER_VERSION);
+
+    /* Pop annex tag */
+    uint8_t annex_tag;
+    annex.read(AsWritableBytes(Span{&annex_tag, 1}));
+    assert(annex_tag == ANNEX_TAG);
+
+    /* Annex record are encoded as:
+     * `VarInt` + 1 bytes: type
+     * `VarInt` OR 0 bytes: length
+     * `length`-bytes: value
+     */
+    while (!annex_stack.empty()) {
+
+        /* `VarInt`-bytes to encode the higher-mask type
+         * 1-byte to encode the lower-mask type
+         * 8192 records with 2 bytes.
+         * 1056767 records with 3 bytes.
+         * 40456331200 records with 4 bytes.
+         * See `VarIntMode` comment in serialize.h for more.
+         */
+        uint64_t nRecordHigherMask = 0;
+        annex >> VARINT(nRecordHigherMask);
+        uint8_t nRecordLowerMask;
+        annex.read(AsWritableBytes(Span{&nRecordLowerMask, 1}));
+
+        const uint64_t nRecordType = nRecordHigherMask * 64 + (nRecordLowerMask & 0x3F);
```

Comment:
Please make constants for masks with explanations what they're for

Reply:
More comments, see the pending question in the BIP.